### PR TITLE
feat(core): add snapshot mode to lifecycle (stop before configDidLoad phase)

### PR DESCRIPTION
**Pull request by killagu**
## Summary

- Add `snapshot?: boolean` option to `EggCoreOptions` and `LifecycleOptions`
- In snapshot mode, lifecycle stops after `didLoad` phase — skips `willReady`/`didReady`/`serverDidReady`
- Foundation for V8 snapshot construction: loads all metadata without creating runtime resources

This is **PR2 of 6** in the V8 startup snapshot series. Independent, no dependencies.

## Changes
- `packages/core/src/egg.ts` — Add `snapshot` option, pass to lifecycle
- `packages/core/src/lifecycle.ts` — Snapshot-aware ready logic (stop after didLoad)

## Test plan
- [x] All 36 core test files pass (396 tests, 25 skipped)
- [x] oxlint --type-aware clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a snapshot startup mode that marks the app ready immediately after initial configuration load and skips subsequent startup lifecycle phases.
* **Tests**
  * Added tests validating snapshot behavior: which startup hooks run or are skipped, readiness timing in snapshot vs normal startup, and cleanup/close behavior.

### CODE REVIEW

File: packages/core/src/egg.ts [hunk 1]
```diff
@@ -33,6 +33,12 @@ export interface EggCoreOptions {
   env?: string;
   /** Skip lifecycle hooks, only trigger loadMetadata for manifest generation */
   metadataOnly?: boolean;
+  /**
+   * When true, the application loads metadata only (plugins, configs, extensions,
+   * services, controllers) without starting servers, timers, or connections.
+   * Used for V8 startup snapshot construction.
+   */
+  snapshot?: boolean;
```
[the remark this answers is not in the document]
Comment (killagu) [reply packages/core/src/egg.ts:43]: Valid point — the JSDoc is intentionally simplified but could be clearer. The key semantic is: **lifecycle stops after `didLoad`, skipping `willReady`/`didReady`/`serverDidReady`**. Plugins that follow Egg conventions (start connections in `willReady`, not `didLoad`) will behave correctly.

Will update the JSDoc to something like:
```
When true, lifecycle stops after didLoad phase.
willReady, didReady, and serverDidReady are skipped.
Used for V8 startup snapshot construction.
```
Thanks for the catch.

File: packages/core/src/lifecycle.ts [hunk 2]
```diff
@@ -69,6 +69,12 @@ export interface LifecycleOptions {
   baseDir: string;
   app: EggCore;
   logger: EggConsoleLogger;
+  /**
+   * When true, the lifecycle stops after didLoad phase completes.
+   * willReady, didReady, and serverDidReady hooks are NOT called.
+   * Used for V8 startup snapshot construction.
```
[the remark this answers is not in the document]
Comment (killagu) [reply packages/core/src/lifecycle.ts]: This is by design. The JSDoc documents what the **lifecycle flow** does automatically — it does not call `serverDidReady`. External callers invoking `triggerServerDidReady()` directly would be doing so intentionally and at their own risk.

Making it a no-op would prevent legitimate use cases where snapshot consumers need to selectively trigger later phases. The current wording ("hooks are NOT called") is accurate for the lifecycle's own behavior. Won't change, but appreciate the thoroughness.

File: packages/core/src/egg.ts [hunk 3]
```diff
@@ -33,6 +33,12 @@ export interface EggCoreOptions {
   env?: string;
   /** Skip lifecycle hooks, only trigger loadMetadata for manifest generation */
   metadataOnly?: boolean;
+  /**
+   * When true, the application loads metadata only (plugins, configs, extensions,
+   * services, controllers) without starting servers, timers, or connections.
+   * Used for V8 startup snapshot construction.
```
[the remark this answers is not in the document]
Comment (killagu) [reply packages/core/src/egg.ts]: The suggested 8-line JSDoc is too verbose for a simple boolean option. The key information developers need is: (1) lifecycle stops after didLoad, (2) later phases are skipped, (3) purpose is snapshot construction.

Will update to a concise version that accurately reflects the behavior without over-documenting edge cases that are better covered in framework docs. See reply to CodeRabbit's similar comment above.

File: packages/core/src/lifecycle.ts [hunk 4]
```diff
@@ -372,6 +378,10 @@ export class Lifecycle extends EventEmitter {
       debug('trigger didLoad end');
       if (err) {
         this.ready(err);
+      } else if (this.options.snapshot) {
+        // In snapshot mode, stop after didLoad — skip willReady/didReady/serverDidReady
+        debug('snapshot mode: skipping willReady, marking ready after didLoad');
+        this.ready(true);
       } else {
         this.triggerWillReady();
```
[the remark this answers is not in the document]
Comment (killagu) [reply packages/core/src/lifecycle.ts]: Agree that tests are needed. Snapshot mode tests will be included in the integration test PR (`feat/snapshot-scripts`, PR #5853) which exercises the full snapshot build + restore flow end-to-end. Adding isolated lifecycle unit tests here would duplicate that coverage. The core change is minimal (3 lines of control flow) and well-understood.

**Review by jerryliang64 — APPROVED** [review]
LGTM
